### ISSUE TITLE
Question: what is max_hits used for?

### ISSUE THREAD

**Issue by Nick-Mazuk** [typesense/typesense#198]
## Description

Why would someone want to use `max_hits`? More specifically, unless you're on page 50 (for instance) and need to raise the `max_hits` to get any results, why would you use it?

Since `per_page` is 10 by default, it seems like the effective max hits will always be 10 * `page` unless you change `per_page`. So on page 1, you'll only ever get 10 results, regardless of `max_hits`.

This is the only reference to this parameter in the docs:

<img width="741" alt="Screen Shot 2021-01-24 at 3 36 23 PM" src="https://user-images.githubusercontent.com/5659171/105647354-ec084f80-5e59-11eb-96d6-399223ff1a05.png">

This raises a few questions:

- Would lowering `max_hits` decrease latency?
- Does `max_hits` affect ranking (e.g., lowering it would reduce latency because Typesense isn't ranking as many documents)?
- What would be the effects of setting `max_width = page * per_page`?
- Is this just a preventative security measure from keeping people from scraping your entire index?

Thanks!

## Metadata

**Typsense Version**: 0.118

**OS**: macOS 11.1

**Comment by jasonbosco** [comment]
@Nick-Mazuk Great questions all around! The context around `max_hits` is that we used to use this param to limit the total number of results that can be fetched from a collection. We deprecated this field in `v0.13.0` because there was an ask to be able to fetch all results. 

Months later, in `v0.18.0` we updated our multi-field search algorithm to take the top-X results from each field's index and "mash them" together to produce the final result set. We wanted a parameter to represent the value of X and decided to name it `max_hits` (unfortunately), which we later realized is quite confusing in hindsight. 

So in `v0.19.0` (coming out in a week or two) we'll actually be doing away with this param altogether and instead dynamically keep expanding top-X based on page number requested.
